plugins/org.polarsys.capella.cybersecurity.sirius.analysis/src/org/polarsys/capella/cybersecurity/sirius/analysis/CybersecurityServices.java

```diff
@@ -241,6 +256,12 @@ public int getAssetColor1Blue(DSemanticDecorator view) {
   public int getAssetColor1Green(DSemanticDecorator view) {
     EObject e = view.getTarget();
     DDiagram diagram = CapellaServices.getService().getDiagramContainer(view);
+    
+    if (e instanceof Part && ((Part) e).getAbstractType() instanceof PhysicalComponent
+        && !isSupportingAssetHighlightedComponent(view)) {
+      return 0;
+    }
+    
     if (diagram != null) {
       int nbOfElements = 0;
       int color = 0;

```

Comment:
getRelatedAssets(e).stream().collect(Collectors.toSet() can remove the to Set part as the getRelatedAssests was modified to contain uniqe elems.